context/llpcCompiler.cpp

```diff
@@ -823,6 +832,170 @@ Result Compiler::BuildShaderModule(
     return result;
 }
 
+Result Compiler::BuildPipelineUsingRelocatableElf(
+    Context*                            pContext,                   // [in] Acquired context
+    ArrayRef<const PipelineShaderInfo*> shaderInfo,                 // [in] Shader info of this pipeline
+    uint32_t                            forceLoopUnrollCount,       // [in] Force loop unroll count (0 means disable)
+    ElfPackage*                         pPipelineElf)               // [out] Output Elf package
+{
+    Result result = Result::Success;
+
+    // Determine whether or not relocatable elf can be used.
+    // We only do relocatable elf if it is a VsPs shader that is given in
+    // spir-v.
+    bool useRelocatableShaders = true;
+    for (uint32_t stage = 0; stage < shaderInfo.size(); ++stage)
+    {
+        if (stage != ShaderStageVertex && stage != ShaderStageFragment)
+        {
+            if ((shaderInfo[stage] != nullptr) && (shaderInfo[stage]->pModuleData != nullptr))
+            {
+                useRelocatableShaders = false;
+            }
+        }
+        else if (shaderInfo[stage] == nullptr || shaderInfo[stage]->pModuleData == nullptr)
+        {
+            useRelocatableShaders = false;
+        }
+    }
+
+    if (useRelocatableShaders && shaderInfo[0] != nullptr)
+    {
+        const ShaderModuleData* pModuleData = reinterpret_cast<const ShaderModuleData*>(shaderInfo[0]->pModuleData);
+        if ((pModuleData != nullptr) && (pModuleData->binType == BinaryType::LlvmBc))
+        {
+          useRelocatableShaders = false;
+        }
+    }
+
+    // For debugging you can add the option |-relocatable-shader-elf-limit=<n>|
+    // to limit the number of time relocatable shaders are used to n.
```

Comment:
This comment would be more helpful at the cl::opt definition.

Reply:
Okay I'll do that.  My plan is to move it there, and remove it from here.  Unless you tell me you want it in both places.

Reply:
Thank you.